lib/mix/lib/mix/tasks/format.ex

```diff
@@ -251,10 +251,12 @@ defmodule Mix.Tasks.Format do
   def formatter_for_file(file, opts \\ []) do
     {dot_formatter, formatter_opts} = eval_dot_formatter(opts)
 
+    prefix = Keyword.get(opts, :root, [])
+
     {formatter_opts_and_subs, _sources} =
-      eval_deps_and_subdirectories(dot_formatter, [], formatter_opts, [dot_formatter])
+      eval_deps_and_subdirectories(dot_formatter, prefix, formatter_opts, [dot_formatter])
```

Comment:
This is broken. `prefix` is meant to be a list and if you pass `:root` as a string, it won't work.

Reply:
I am on it.